### PR TITLE
Add named configurations

### PR DESCRIPTION
**Pull request by carabasdaniel**
Depends on https://github.com/aserto-dev/topaz/pull/256

### CODE REVIEW

**Review by BogdanIrimie — DISMISSED** [review]

File: cmd/topaz/main.go [hunk 1]
```diff
@@ -34,17 +45,19 @@ func main() {
...
-		fmt.Fprintln(os.Stderr, err.Error())
-		os.Exit(1)
-	}
-
 	if err := kongCtx.Run(ctx); err != nil {
 		kongCtx.FatalIfErrorf(err)
 	}
+
+	// only save on config change
+	if app.Contains(kongCtx.Args, "configure") || app.Contains(kongCtx.Args, "-c") || app.Contains(kongCtx.Args, "--config") {
```
Comment (ronenh) [inline cmd/topaz/main.go:57]: The `topaz version` command has a `-c` flag too. I don't think we want to save configuration in that case.

File: pkg/app/topaz.go [hunk 2]
```diff
@@ -267,7 +267,7 @@ func mapToGRPCPorts(api map[string]*builder.API) map[string]services {
 	return portMap
 }
 
-func contains[T comparable](slice []T, item T) bool {
+func Contains[T comparable](slice []T, item T) bool {
```
Comment (ronenh) [inline pkg/app/topaz.go:270]: We should replace this function with [`lo.Contains`](https://github.com/samber/lo?tab=readme-ov-file#contains)

File: pkg/cli/cc/cc.go [hunk 3]
```diff
@@ -14,7 +16,16 @@ import (
 type CommonCtx struct {
 	Context context.Context
 	UI      *clui.UI
-	NoCheck bool
+	Config  *CLIConfig
+}
+
+type CLIConfig struct {
+	NoCheck           bool
+	DefaultConfigFile string
```
Comment (ronenh) [inline pkg/cli/cc/cc.go:24]: I would rename this field to `TopazConfigFile`. "Default" makes it seem like there are other, non-default, files but a CLIConfig refers to exactly one topaz config file.

File: pkg/cli/cmd/configure.go [hunk 4]
```diff
@@ -15,10 +15,11 @@ type ConfigureCmd struct {
...
 	LocalPolicyImage  string `short:"l" help:"local policy image name"`
 	Resource          string `short:"r" help:"resource url"`
-	Stdout            bool   `short:"p" help:"generated configuration is printed to stdout but not saved"`
+	Stdout            bool   `short:"p" help:"print to stdout"`
 	EdgeDirectory     bool   `short:"d" help:"enable edge directory" default:"false"`
-	Force             bool   `flag:"" default:"false" short:"f" help:"forcefully create configuration"`
-	EnableDirectoryV2 bool   `flag:"" name:"enable-v2" default:"false" help:"enable directory version 2 services for backward compatibility"`
+	Force             bool   `flag:"" default:"false" short:"f" required:"false" help:"skip confirmation prompt"`
+	EnableDirectoryV2 bool   `flag:"" name:"enable-v2" hidden:"" default:"true" help:"enable directory version 2 services for backwards compatibility"`
+	ConfigFile        string `name:"config" json:"config,omitempty" default:"config.yaml" short:"c" help:"topaz configuration file"`
```
Comment (ronenh) [inline pkg/cli/cmd/configure.go:22]: What do you think about calling this field `ConfigName` and lose the `.yaml` extension.
When we read and write configs from/to disk we'll include the `.yaml` suffix but when users create, list, or use configurations, they'll just specify the file name.
For example, if I have a config file called `peoplefinder.yaml`, the configuration name would be just `peoplefinder`.

When calling `topaz configure` we could default to using the policy name as the config name, unless the caller explicitly sets the `ConfigName` param to override it.
So we could run
```bash
topaz configure -r ghcr.io/aserto-policies/policy-peoplfinder:latest -n peoplefinder
```
and generate `peoplefinder.yaml`.

Comment (carabasdaniel) [reply pkg/cli/cmd/configure.go:22]: I'll put this change in a separate PR so it's easier to review

**Review by ronenh — DISMISSED** [review]

File: pkg/cli/cc/cc.go [hunk 5]
```diff
@@ -16,10 +19,25 @@ import (
...
+	Config  *CLIConfig
+}
+
+type CLIConfig struct {
+	NoCheck           bool
+	TopazConfigFile   string
+	ContainerName     string
+	ContainerRegistry string
+	ContainerImage    string
+	ContainerTag      string
```
Comment (gertd) [inline pkg/cli/cc/cc.go:40]: Should `ContainerPlatform` be part of this list as it is a settable parameter like the others? 
Besides consistency with the other parameters, I can see this being useful when insisting to use amd64 images on a Mac arm64 machine.

File: pkg/cli/cmd/cli.go [hunk 6]
```diff
@@ -1,44 +1,71 @@
...
+	"github.com/aserto-dev/topaz/pkg/cli/cmd/directory"
 )
 
-type FormatVersion int
+type SaveContext bool
 
 const (
-	V2 FormatVersion = 2
-	V3 FormatVersion = 3
+	CLIConfigurationFile = "cli_config.json"
```
Comment (gertd) [inline pkg/cli/cmd/cli.go:16]: Suggest renaming this to `config.json` or `topaz.json`.
The fact it resides in XDG_CONFIG_HOME already reveals its nature. 
The name `cli_*` feels odd.

**Review by gertd — APPROVED** [review]

**Pull request merged**